src/hooks/useStageHandler.ts

```diff
@@ -90,6 +94,21 @@ export const useStageHandler = () => {
         );
         await tx.wait();
         toast.success("Your transaction was successful");
+        if (id) {
+          const contract = await getSmartContractWithProvider(
```

Comment:
put TODO: we should save the factory address in memory instead of keep getting it every time.